server/core/src/rodsServer.cpp

```diff
@@ -394,41 +395,46 @@ int main(int argc, char** argv)
     snprintf(local_addr.sun_path, sizeof(local_addr.sun_path), "%s", agent_factory_socket_file);
 
     ix::log::server::info("Forking agent factory ...");
+    ix::cron::cron_builder agent_watcher;
+    auto start_agent_server = [&](std::any& _) {
+        int status;
+        int w = waitpid(agent_spawning_pid, &status, WNOHANG);
+        if( w != 0 ) {
+            ix::log::server::info("Restarting agent spawner");
+            auto new_pid = fork();
+            if( new_pid == 0 ) {
+                ProcessType = AGENT_PT;
+                runIrodsAgentFactory(local_addr);
```

Comment:
The original code returned the result of this function call. We need to investigate how to maintain the original behavior here.

Reply:
where did it expose it?

Didn't it just end up getting gobbled up by the fork without it ever being recorded?

Reply:
though, if it's about the process needing to exit with a specifc code, I guess that `exit(runIrodsAgentFactory(local_addr));` would probably suffice

Reply:
`exit` may work, but I'm nervous that it may not do the full job. Some code in `main` expects to run during the call to destructors and `exit` doesn't guarantee that.

see https://en.cppreference.com/w/cpp/utility/program/exit

Reply:
Does the stack get unwound when the process aborts or is killed any other way?

Reply:
Okay, so I can cause the destructors to fire off if I throw, for example, the return code from the forked process.

Reply:
So that doesn't work because there are plenty of `catch (...)` are in the way, it results in far too many agent factories being spawned. I think